upstream_utils/README.md

```diff
@@ -64,3 +64,45 @@ Modify the version number in the call to `setup_upstream_repo()` in
 ```bash
 ./update_lib.py
 ```
+
+## Adding patch to thirdparty library
+
+The example below will add a new patch file to a hypothetical library called
+`lib`.
+
+Start in the `upstream_utils` folder. Restore the original repo.
+```bash
+./update_lib.py
+```
+
+Navigate to the repo.
+```bash
+cd /tmp/lib
```

Comment:
```suggestion
cd /tmp/<lib>
```
On my mac this was `/private/var/folders/2t/jzczznpx033c6tbrdt24ldhw0000gn/T/llvm-project`

Reply:
Is that actually a persistent name? Given how it looks, I wouldn't be surprised if `tempfile.gettempdir()` returned a different random hex string each time the script is run on macOS, causing a new repo to be cloned during each run.

Reply:
> Is that actually a persistent name? 

No.